app/services/data_stage/process_school_changes.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module DataStage
+  class ProcessSchoolChanges < ::BaseService
+    include GiasTypes
+
+    UNHANDLED_ATTRIBUTES = %w[
+      administrative_district_code
+      administrative_district_name
+      school_type_code
+      school_type_name
+      section_41_approved
+      ukprn
+    ].freeze
+
+    def call
+      DataStage::SchoolChange.unhandled.status_changed.find_each do |change|
+        next if has_unhandled_attributes?(change.attribute_changes)
+
+        if change.attribute_changes.key? "school_status_code"
+          handle_status_code_change(change)
+        end
+      end
+    end
+
+  private
+
+    def has_unhandled_attributes?(attributes)
+      (UNHANDLED_ATTRIBUTES & attributes.keys).any?
+    end
+
+    def handle_status_code_change(change)
+      from_code = change.attribute_changes["school_status_code"]&.first
+      to_code = change.attribute_changes["school_status_code"]&.last
+
+      if open_status_code?(from_code) && !open_status_code?(to_code)
+        close_school(change)
+      elsif !open_status_code?(from_code) && open_status_code?(to_code)
+        open_school(change)
+      else
+        # just sync then
+        change.school.create_or_sync_counterpart!
+      end
+    end
+
+    def open_school(change)
+      school = change.school
+
+      # is there a predecessor
+      last_link = school.school_links.order(created_at: :asc).last
```

Comment:
If we call the variable `predecessor` then the comment is unnecessary, right?

Reply:
Unless the comment is lying, and the actual check for whether there is a predecessor is
```
last_link.link_type == "Predecessor"
```
line.

Reply:
`predecessor` is used later but fair point about the comment

Reply:
Moving this class to a separate PR